### ISSUE TITLE
Problem accessing/deleting restored mod_bigbluebuttonbn instances

### ISSUE THREAD

**Issue by logan-reynolds** [blindsidenetworks/moodle-mod_bigbluebuttonbn#102]
Because the activity is course-specific, the given activity and corresponding room do not seem to exist in a restored instance of the course.  Activity is still accessible and can be edited/deleted in the original course, but restored copy of course backup has an error when accessing the copied mod instance with the following error:
```
SELECT cm.*, m.name, md.name AS modname FROM {course_modules} cm JOIN {modules} md ON md.id = cm.module JOIN {bigbluebuttonbn} m ON m.id = cm.instance WHERE cm.id = :cmid AND md.name = :modulename AND cm.course = :courseid [array ( 'cmid' => 52, 'modulename' => 'bigbluebuttonbn', 'courseid' => '11', )] Error code: invalidrecordunknown
Stack trace:

* line 1687 of /lib/dml/moodle_database.php: dml_missing_record_exception thrown
* line 1332 of /lib/datalib.php: call to moodle_database->get_record_sql()
* line 55 of /course/rest.php: call to get_coursemodule_from_id()
```
The corresponding activity should be copied as a valid one in the database so it can be accessed/removed from the restored instance, or at least delete the activity id from mdl_course_modules if the record doesn't exist in the mdl_bigbluebuttonbn table

**Comment by logan-reynolds** [comment]
Hello just checking back on this one to see if it's able to be reproduced, and whether you need any further information.

**Comment by geoffhorsfall** [comment]
Checking in again on this issue from logan-reynolds

**Comment by jfederico** [comment]
Thanks, we are focused on the new release (2.3) right now.  https://tracker.moodle.org/browse/CONTRIB-7625?jql=labels%20%3D%202.3
Would you be able to check if that bug is still in the code we have in master?

**Comment by logan-reynolds** [comment]
this seems to be resolved in the latest version 2017101018 release 2.2.10 from Moodle.org, so closing this as resolved.
